pkg/kubernetes/deploy.js

```diff
@@ -0,0 +1,373 @@
+                    for (var i = 0; i < jdata.items.length; i++) {
+                        var ent_json = jdata.items[i];
+                        //console.log(ent_json)
+                        if (ent_json.kind === client.SERVICE) {
+                            services.push(ent_json);
+                        } else if (ent_json.kind === client.POD) {
+                            pods.push(ent_json);
+                        } else if (ent_json.kind === client.RC) {
+                            rcs.push(ent_json);
+                        } else if (ent_json.kind === client.NS) {
+                            namespaces.push(ent_json);
+                        }
+                    }
+                }
+            } else {
+                var text = "Unable to Read the file.Please check the json file. ";
+                file_note.show();
+                file_note_details.text(text);
+                return;
+            }
+
+            create_everything(namespace, services, rcs, pods)
+                .progress(function(code, response, n, total) {
+                    //console.log(" " + code + "  " + response + "  " + n + " / " + total);
+                })
+                .done(function() {
+                    /* code gets run when everything is created */
+                    var btn = $('#deploy-app-stop');
+                    btn.removeClass('btn-default');
+                    btn.addClass('btn-primary');
+                    btn.text("OK");
+                    hide_progress_message();
+                    is_deploying.parent().prepend($(Mustache.render(deploy_notification_success)));
+
+                })
+                .fail(function(ex, response) {
+                    /* ex containst the failure */
+                    var jdata = JSON.parse(response);
+                    hide_progress_message();
+                    var context = {};
+                    is_deploying.parent().prepend($(Mustache.render(deploy_notification_failure, $.extend(context, jdata))));
+
+                });
+        }
+
+        function create_everything(namespace, services, rcs, pods) {
+            var ns_json = '{"apiVersion":"v1beta3","kind":"Namespace","metadata":{"name": "' + namespace + '"}}';
+            var tasks = [];
+
+            tasks.push(["namespace", [ns_json], client.create_ns]);
+
+            for (var serv in services) {
+                tasks.push([services[serv].metadata.name, [namespace, JSON.stringify(services[serv])], client.create_service]);
+            }
+
+            for (var rc in rcs) {
+                tasks.push([rcs[rc].metadata.name, [namespace, JSON.stringify(rcs[rc])], client.create_replicationcontroller]);
+            }
+
+            for (var p in pods) {
+                tasks.push([pods[p].metadata.name, [namespace, JSON.stringify(pods[p])], client.create_pod]);
+            }
+
+            var deferred = $.Deferred();
+            var total = tasks.length;
+
+            function step(qtasks) {
```

Comment:
Why do you pass in ```qtasks``` as an argument? Why don't you just access ```tasks``` from the outer scope?